From 4a5cedb9e04913068614f48389fae32240a482dd Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 12 Nov 2015 15:40:00 -0800
Subject: [PATCH] BSEAV/linux/driver/usermode/Makefile: support separate kernel
 source and builds dirs

Also use the same order for kernel header include paths as KBuild does.

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 BSEAV/linux/driver/usermode/Makefile | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/BSEAV/linux/driver/usermode/Makefile b/BSEAV/linux/driver/usermode/Makefile
index 6a3ec7a..8d7f958 100644
--- a/BSEAV/linux/driver/usermode/Makefile
+++ b/BSEAV/linux/driver/usermode/Makefile
@@ -113,7 +113,6 @@ vpath %.c ${BCM_OBJ_ROOT}/interrupt_table
 
 BCM_DRIVER := $(BCM_OBJ_PATH)/bcmdriver.ko
 BCM_DRIVER_CFLAGS += -DKBUILD_MODNAME=bcmdriver -DLINUX -D__KERNEL__ -DMODULE -mlong-calls
-BCM_DRIVER_CFLAGS += -I$(LINUX)/include -I$(LINUX)/arch/$(B_REFSW_LINUX_ARCH)/include
 ifeq ($(LINUX_VER_GE_3),y)
 BCM_DRIVER_CFLAGS += -include linux/kconfig.h
 else
@@ -129,7 +128,21 @@ BCM_DRIVER_SRCS += \
 	bcm_driver.c
 
 ifeq ($(B_REFSW_LINUX_ARCH),arm)
+
+ifneq ($(STAGING_KERNEL_BUILDDIR),)
+BCM_DRIVER_CFLAGS += \
+	-I$(LINUX)/arch/arm/include \
+	-I$(STAGING_KERNEL_BUILDDIR)/arch/arm/include/generated \
+	-I$(LINUX)/include \
+	-I$(STAGING_KERNEL_BUILDDIR)/include \
+	-I$(LINUX)/arch/arm/include/uapi \
+	-I$(STAGING_KERNEL_BUILDDIR)/arch/arm/include/generated/uapi \
+	-I$(LINUX)/include/uapi \
+	-I$(STAGING_KERNEL_BUILDDIR)/include/generated/uapi
+else
+BCM_DRIVER_CFLAGS += -I$(LINUX)/include -I$(LINUX)/arch/$(B_REFSW_LINUX_ARCH)/include
 BCM_DRIVER_CFLAGS += -I$(LINUX)/arch/arm/include -I$(LINUX)/arch/arm/include/generated -I$(LINUX)/arch/arm/include/uapi -I$(LINUX)/arch/arm/include/generated/uapi -I$(LINUX)/include/uapi -I$(LINUX)/include/generated/uapi
+endif
 
 BCM_DRIVER_SRCS_AS = bcmdriver_arm_v7.S
 
@@ -138,6 +151,17 @@ ifeq ($(BRCMSTB_MEGA_BARRIER),brcmstb_mega_barrier)
 BCM_DRIVER_CFLAGS += -DUSE_BRCMSTB_MEGA_BARRIER
 endif
 
+else ifeq ($(B_REFSW_LINUX_ARCH),mips)
+
+ifneq ($(STAGING_KERNEL_BUILDDIR),)
+BCM_DRIVER_CFLAGS += \
+	-I$(LINUX)/arch/mips/include \
+	-I$(STAGING_KERNEL_BUILDDIR)/arch/mips/include/generated \
+	-I$(STAGING_KERNEL_BUILDDIR)/include \
+	-I$(LINUX)/include
+else
+BCM_DRIVER_CFLAGS += -I$(LINUX)/include -I$(LINUX)/arch/$(B_REFSW_LINUX_ARCH)/include
+endif
 endif
 
 ICM_CLI_SUPPORT ?= y
-- 
1.9.1

